### PR TITLE
Fix for creating syntax tree based generators for OpenAPI

### LINKED ISSUE #92: Create syntax tree based generators for OpenAPI
**Description:**
Purpose of this is to create openapi generators based on the syntax tree

- [x] Implementation for  imports
- [x] Implementation for lisnteners
- Implementation serivceDeclaratoinNode
  - [x] Implementation for basic resource function
  - [x] Implementation  for resource function with path parameters
  - [x] Implementation  for resource function with query parameters
  - [x] Implementation  for resource function with query parameters with array type
  - [x] Implementation  for resource function with header parameters
  - [x] Implementation  for resource function with request body parameters
  - [x] Implementation  for resource function with reponse type with 200 status code
  - [x] Implementation  for resource function with reponse type with different status code
  - [x] Implementation  for multiple operations
  - [x] Implementation  for multiple paths 
- [x] Fix syntax tree with formatting

 Related documentation [https://docs.google.com/document/d/1ubuKZ9GXRP0_b_vy8NLKIvdbQsyhTIcRql9HPmITA-I/edit?ts=603730bd#](https://docs.google.com/document/d/1ubuKZ9GXRP0_b_vy8NLKIvdbQsyhTIcRql9HPmITA-I/edit?usp=sharing)

**Suggested Labels:**

**Suggested Assignees:**

**Affected Product Version:**

**OS, DB, other environment details and versions:**    

**Steps to reproduce:**

**Related Issues:**

### LINKED ISSUE #117: Create syntax tree based generators with ballerina schema for OpenAPI components
**Description:**
$Subject

**Suggested Labels:**

**Suggested Assignees:**

**Affected Product Version:**

**OS, DB, other environment details and versions:**    

**Steps to reproduce:**

**Related Issues:**

### PR DESCRIPTION
**Pull request by lnash94**
Fix #117 

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: openapi-cli/src/main/java/org/ballerinalang/generators/BallerinaSchemaGenerator.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.generators;
+
+/**
+ *
```
Comment (hemikak) [inline openapi-cli/src/main/java/org/ballerinalang/generators/BallerinaSchemaGenerator.java:22]: Add documentation

File: openapi-cli/src/main/java/org/ballerinalang/generators/BallerinaServiceGenerator.java [hunk 2]
```diff
@@ -0,0 +1,1043 @@
...
+    private static Token closeBraceToken = AbstractNodeFactory.createIdentifierToken("}");
+    private static Token openSBracketToken = AbstractNodeFactory.createIdentifierToken("[");
+    private static Token closeSBracketToken = AbstractNodeFactory.createIdentifierToken("]");
+    private static Token colon = AbstractNodeFactory.createIdentifierToken(":");
+    private static Token semicolonToken = AbstractNodeFactory.createIdentifierToken(";");
+    // Add basicLiteralNode
+    private static MinutiaeList leading = AbstractNodeFactory.createEmptyMinutiaeList();
+    private static Minutiae whitespace = AbstractNodeFactory.createWhitespaceMinutiae(" ");
+    private static MinutiaeList trailing = AbstractNodeFactory.createMinutiaeList(whitespace);
+    private static Token questionMark = NodeFactory.createIdentifierToken("?");
```
Comment (hemikak) [inline openapi-cli/src/main/java/org/ballerinalang/generators/BallerinaServiceGenerator.java:124]: Make them final if these variables doesnt change value.

File: openapi-cli/src/main/java/org/ballerinalang/generators/BallerinaServiceGenerator.java [hunk 3]
```diff
@@ -0,0 +1,1043 @@
...
+@Nonnull
```
Comment (hemikak) [inline openapi-cli/src/main/java/org/ballerinalang/generators/BallerinaServiceGenerator.java:125]: Do we need this annotation ? If needed, please fix indentation

File: openapi-cli/src/main/java/org/ballerinalang/generators/BallerinaSchemaGenerator.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.generators;
+
+/**
+ * This util class for generate record syntax tree for component schemas.
+ */
+public class BallerinaSchemaGenerator {
```
Comment (hemikak) [inline openapi-cli/src/main/java/org/ballerinalang/generators/BallerinaSchemaGenerator.java:24]: Empty class ?

**Review by hemikak — APPROVED** [review]

**Review by hevayo — APPROVED** [review]

**Pull request merged**